### ISSUE TITLE
Direct usage of global window object 

### ISSUE THREAD

**Issue by davidkudera** [LPology/Simple-Ajax-Uploader#124]
One of the main reasons why people should use commonjs or amd is to be sure, that their object and also objects from foreign libraries does not mess up the global scope and does not use the global object directly.

It's great that this library now supports these techniques, but it [violates](https://github.com/LPology/Simple-Ajax-Uploader/blob/84fc0c456451b7a6fba1032d0bd667f1fd87f1ac/SimpleAjaxUploader.js#L23) this "rule" too a little bit. 

Some examples of the "right" aproach:
* [nette/forms](https://github.com/nette/forms/blob/cc02e729bf1e8dbeae80687ddcadb83ed992c04a/src/assets/netteForms.js#L8-L28)
* [nprogress](https://github.com/rstacruz/nprogress/blob/254266ea50c220f6507fa3f2978fc530147c7f06/nprogress.js#L4-L15)

Also there is this nice article: http://ifandelse.com/its-not-hard-making-your-library-support-amd-and-commonjs/

**Comment by LPology** [comment]
Thanks for the feedback. Would this be better?

http://pastebin.com/uxTYuT2S

Modeled after the nette/forms link.

**Comment by LPology** [comment]
Also, do you mind pointing to some specific instances of where you see potential issues? Any help would be great.

**Comment by davidkudera** [comment]
I think that would be great :+1: 

The thing is that with such a code, your library would be completly isolated from the global scope. 

Imagine for example tests, you don't need to work with global window, you can simply use some mocked window object only with needed properties or functions.

Or if developer install some other library which also uses the `ss` variable, either yours or the other library will not work properly (or at all)...

Simply put, no one can now break your library from outside and also your library can't break any other (only itself :wink:).

**Comment by LPology** [comment]
Forgot to close this out before. Thanks for your input.
